junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExecutableInvoker.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2015-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+
+import org.apiguardian.api.API;
+
+@API(status = EXPERIMENTAL, since = "5.9")
+public interface ExecutableInvoker {
+
+	Object invoke(Executable executable, Object target);
+
+	<T> T invoke(Constructor<T> constructor, Object outerInstance);
+
+}
```

Comment:
I was thinking about something like this:

```suggestion
public interface ExecutableInvoker {

	// static method
	Object invoke(Method method);

	// instance method
	Object invoke(Method method, Object target);

	// top-level constructor
	<T> T invoke(Constructor<T> constructor);

	// nested constructor
	<T> T invoke(Constructor<T> constructor, Object enclosingInstance);

}
```

WDYT?

Reply:
I didn't know whether to detail all use cases or not (did the constructor one because typing), but I agree that it looks more user-friendly this way! 👍

Reply:
Yeah, my goal is to make it easy for the user to comprehend.

The `default` methods are just for convenience, but I like them because they allow us to hide the technical details of how we actually implement it behind the scenes.

Glad you like the proposal. 😄 

Let's see what others have to say.